Create sick_tim571_2050001.launch

sacuar opened the pull request:
This launch file can be used directly to connect TIM571 devices through TCP/IP.

sacuar committed:
Create sick_tim571_2050001.launch

This launch file can be used directly to connect TIM571 devices through TCP/IP.

mintar commented:
Thanks @sacuar ! In principle, I think it makes sense to add a launch file to make it clear to users which parameters to use for the scanner. However, there are a couple of issues before we can merge:
- The only important changes you made are:
  - change the node name
  - increase the `range_max` parameter to 30 m.
  
  But your commit contains lots of other changes as well, which are unnecessary. So we should make this new launch file a direct copy of sick_tim551_2050001.launch with only those two things changed (and of course moving the range_max param out of the comment block).
- Are you sure that the max range is 30 m? I couldn't find any spec sheet online yet, but the [only other source](http://robotsforroboticists.com/sick-tim551-lidar-review/) says 25 m. Do you have a spec sheet that you could send me?
- The number 2050001 is the model number of the TiM551. Could you please tell me the model number and part number of your laser scanner? It should be on the side of the device. Then we should also rename the launch file appropriately.

sacuar commented:
That's what the marketing manager of SICK in Germany "Daniel Perez" ensured me about. And I got his contact through CEO of willow garage "Brian Gerkey". Seems like it has got a better range! Or, maybe it was a rough approximation :) Although I have not scaled the measurement.I will write you later if I find the documentation. The new sensor has a resolution of 0.333 degrees which was 1 degree in previous versions. And this can be changed back to 1 degree by using as median filter. The device details :  TIM571-2050101    P/N : 1075091   S/N 15160296 DC-9-28V /3W

sacuar committed:
Update and rename sick_tim571_2050001.launch to sick_tim571_2050101.launch

Changed the launch to directly work with Tim571 through Ethernet.Only needs to change host IP now in the <param name>  "hostname" according user's network.

sacuar commented:
Actually, I thought myself that having a different launch file is more feasible ,because I work with different devices on the same computer. So when I change the parameters then it becomes tiring to change the files and compile again and again.I think other users would think the same.

mintar commented:
Thanks for the info!

One more thing that would be nice: Could you record a short 20 second bag file of the datagram topic for me (just so I can debug without having the device ready)?

How to do that:
- edit your launch file to set the `publish_datagram` param to "True" (and move it out of the comment block)
- launch it
- run: `rosbag record --bz2 --duration=20 /datagram /scan`
- send the resulting *.bag file to me
- set `publish_datagram` to "False" again

---

About this pull request: I've pushed a cleaned up version of your commit, so we should be fine now. I'm not quite sure what you mean by your last comment; the launch files provided in the package can only serve as examples, if you want to change something (like switch from USB to TCP, or change the IP address), it's easiest to just copy the launch file to your own package and modify it there.

sacuar commented:
@mintar I have sent you the bag file in your mail!

mirzashah commented:
@mintar @sacuar  I was also getting a warning about time increment until I set it to the recommended value. 

```
  <param name="range_max" type="double" value="25.0" />
  <param name="time_increment" type="double" value="0.000185185" />
```

mintar commented:
Good catch, @mirzashah ! I could reproduce the warning using @sacuar 's bagfile. Fixed in cf306dc2.
